File: adaptive-arp-api-objc/adaptive-arp-api-objc/ARP/Email.m
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     *
 *
 * =====================================================================================================================
 */

//
//  source: ../../../adaptive-arp-api-java/arp-api-specs/src/main/java/me/adaptive/arp/api/Email.java
//
//

#include "Email.h"
#include "IOSObjectArray.h"

@implementation ARPEmail

- (id)init {
  return [super init];
}

- (id)initWithARPEmailAddressArray:(IOSObjectArray *)toRecipients
          withARPEmailAddressArray:(IOSObjectArray *)ccRecipients
          withARPEmailAddressArray:(IOSObjectArray *)bccRecipients
        withARPAttachmentDataArray:(IOSObjectArray *)attachmentData
                      withNSString:(NSString *)messageBody
                      withNSString:(NSString *)messageBodyMimeType
                      withNSString:(NSString *)subject {
  if (self = [super init]) {
    self->toRecipients_ = toRecipients;
    self->ccRecipients_ = ccRecipients;
    self->bccRecipients_ = bccRecipients;
    self->attachmentData_ = attachmentData;
    self->messageBody_ = messageBody;
    self->messageBodyMimeType_ = messageBodyMimeType;
    self->subject_ = subject;
  }
  return self;
}

- (id)initWithARPEmailAddressArray:(IOSObjectArray *)toRecipients
                      withNSString:(NSString *)subject
                      withNSString:(NSString *)messageBody {
  if (self = [super init]) {
    self->toRecipients_ = toRecipients;
    self->messageBody_ = messageBody;
    self->subject_ = subject;
  }
  return self;
}

- (IOSObjectArray *)getToRecipients {
  return toRecipients_;
}

- (void)setToRecipientsWithARPEmailAddressArray:(IOSObjectArray *)toRecipients {
  self->toRecipients_ = toRecipients;
}

- (IOSObjectArray *)getCcRecipients {
  return ccRecipients_;
}

- (void)setCcRecipientsWithARPEmailAddressArray:(IOSObjectArray *)ccRecipients {
  self->ccRecipients_ = ccRecipients;
}

- (IOSObjectArray *)getBccRecipients {
  return bccRecipients_;
}

- (void)setBccRecipientsWithARPEmailAddressArray:(IOSObjectArray *)bccRecipients {
  self->bccRecipients_ = bccRecipients;
}

- (IOSObjectArray *)getAttachmentData {
  return attachmentData_;
}

- (void)setAttachmentDataWithARPAttachmentDataArray:(IOSObjectArray *)attachmentData {
  self->attachmentData_ = attachmentData;
}

- (NSString *)getMessageBody {
  return messageBody_;
}

- (void)setMessageBodyWithNSString:(NSString *)messageBody {
  self->messageBody_ = messageBody;
}

- (NSString *)getMessageBodyMimeType {
  return messageBodyMimeType_;
}

- (void)setMessageBodyMimeTypeWithNSString:(NSString *)messageBodyMimeType {
  self->messageBodyMimeType_ = messageBodyMimeType;
}

- (NSString *)getSubject {
  return subject_;
}

- (void)setSubjectWithNSString:(NSString *)subject {
  self->subject_ = subject;
}

- (void)copyAllFieldsTo:(ARPEmail *)other {
  [super copyAllFieldsTo:other];
  other->attachmentData_ = attachmentData_;
  other->bccRecipients_ = bccRecipients_;
  other->ccRecipients_ = ccRecipients_;
  other->messageBody_ = messageBody_;
  other->messageBodyMimeType_ = messageBodyMimeType_;
  other->subject_ = subject_;
  other->toRecipients_ = toRecipients_;
}

+ (J2ObjcClassInfo *)__metadata {
  static J2ObjcMethodInfo methods[] = {
    { "init", "Email", NULL, 0x1, NULL },
    { "initWithARPEmailAddressArray:withARPEmailAddressArray:withARPEmailAddressArray:withARPAttachmentDataArray:withNSString:withNSString:withNSString:", "Email", NULL, 0x1, NULL },
    { "initWithARPEmailAddressArray:withNSString:withNSString:", "Email", NULL, 0x1, NULL },
    { "getToRecipients", NULL, "[Lme.adaptive.arp.api.EmailAddress;", 0x1, NULL },
    { "setToRecipientsWithARPEmailAddressArray:", "setToRecipients", "V", 0x1, NULL },
    { "getCcRecipients", NULL, "[Lme.adaptive.arp.api.EmailAddress;", 0x1, NULL },
    { "setCcRecipientsWithARPEmailAddressArray:", "setCcRecipients", "V", 0x1, NULL },
    { "getBccRecipients", NULL, "[Lme.adaptive.arp.api.EmailAddress;", 0x1, NULL },
    { "setBccRecipientsWithARPEmailAddressArray:", "setBccRecipients", "V", 0x1, NULL },
    { "getAttachmentData", NULL, "[Lme.adaptive.arp.api.AttachmentData;", 0x1, NULL },
    { "setAttachmentDataWithARPAttachmentDataArray:", "setAttachmentData", "V", 0x1, NULL },
    { "getMessageBody", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setMessageBodyWithNSString:", "setMessageBody", "V", 0x1, NULL },
    { "getMessageBodyMimeType", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setMessageBodyMimeTypeWithNSString:", "setMessageBodyMimeType", "V", 0x1, NULL },
    { "getSubject", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setSubjectWithNSString:", "setSubject", "V", 0x1, NULL },
  };
  static J2ObjcFieldInfo fields[] = {
    { "toRecipients_", NULL, 0x2, "[Lme.adaptive.arp.api.EmailAddress;", NULL,  },
    { "ccRecipients_", NULL, 0x2, "[Lme.adaptive.arp.api.EmailAddress;", NULL,  },
    { "bccRecipients_", NULL, 0x2, "[Lme.adaptive.arp.api.EmailAddress;", NULL,  },
    { "attachmentData_", NULL, 0x2, "[Lme.adaptive.arp.api.AttachmentData;", NULL,  },
    { "messageBody_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "messageBodyMimeType_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "subject_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
  };
  static J2ObjcClassInfo _ARPEmail = { "Email", "me.adaptive.arp.api", NULL, 0x1, 17, methods, 7, fields, 0, NULL};
  return &_ARPEmail;
}

@end

File: adaptive-arp-api-specs/src/me/adaptive/arp/api/Email.java
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     * Ferran Vila Conesa
 *                 <http://github.com/fnva>
 *                 <http://twitter.com/ferran_vila>
 *                 <mailto:ferran.vila.conesa@gmail.com>
 *
 *     * Aryslan
 *                 <http://github.com/Aryslan>
 *                 <http://twitter.com/Aryslan>
 *                 <mailto:ddbc@gft.com>
 *
 *     * Francisco Javier Martin Bueno
 *                 <https://github.com/kechis>
 *                 <mailto:kechis@gmail.com>
 *
 * =====================================================================================================================
 */

package me.adaptive.arp.api;

/**
 * Structure representing the data elements of an email.
 *
 * @author Francisco Javier Martin Bueno
 * @since ARP 2.0
 */
public class Email extends APIBean {

    /**
     * Array of Email recipients
     */
    private EmailAddress[] toRecipients;

    /**
     * Array of Email Carbon Copy recipients
     */
    private EmailAddress[] ccRecipients;

    /**
     * Array of Email Blind Carbon Copy recipients
     */
    private EmailAddress[] bccRecipients;

    /**
     * Array of attatchments
     */
    private EmailAttachmentData[] emailAttachmentData;

    /**
     * Message body
     */
    private String messageBody;

    /**
     * Message body mime type
     */
    private String messageBodyMimeType;

    /**
     * Subject of the email
     */
    private String subject;

    /**
     * Default constructor
     *
     * @since ARP 2.0
     */
    public Email() {
    }

    /**
     * Constructor used by the implementation
     *
     * @param toRecipients array of recipients
     * @param subject      of the email
     * @param messageBody  body of the email
     * @since ARP 2.0
     */
    public Email(EmailAddress[] toRecipients, String subject, String messageBody) {
        this.toRecipients = toRecipients;
        this.messageBody = messageBody;
        this.subject = subject;
    }

    /**
     * Constructor used by the implementation
     *
     * @param toRecipients        array of recipients
     * @param ccRecipients        array of cc recipients
     * @param bccRecipients       array of bcc recipients
     * @param emailAttachmentData array of attatchments
     * @param messageBody         body of the email
     * @param messageBodyMimeType mime type of the body
     * @param subject             of the email
     * @since ARP 2.0
     */
    public Email(EmailAddress[] toRecipients, EmailAddress[] ccRecipients, EmailAddress[] bccRecipients, EmailAttachmentData[] emailAttachmentData, String messageBody, String messageBodyMimeType, String subject) {
        this.toRecipients = toRecipients;
        this.ccRecipients = ccRecipients;
        this.bccRecipients = bccRecipients;
        this.emailAttachmentData = emailAttachmentData;
        this.messageBody = messageBody;
        this.messageBodyMimeType = messageBodyMimeType;
        this.subject = subject;
    }

    /**
     * Returns the array of recipients
     *
     * @return toRecipients array of recipients
     * @since ARP 2.0
     */
    public EmailAddress[] getToRecipients() {
        return toRecipients;
    }

    /**
     * Set the array of recipients
     *
     * @param toRecipients array of recipients
     * @since ARP 2.0
     */
    public void setToRecipients(EmailAddress[] toRecipients) {
        this.toRecipients = toRecipients;
    }

    /**
     * Returns the array of recipients
     *
     * @return ccRecipients array of cc recipients
     * @since ARP 2.0
     */
    public EmailAddress[] getCcRecipients() {
        return ccRecipients;
    }

    /**
     * Set the array of recipients
     *
     * @param ccRecipients array of cc recipients
     * @since ARP 2.0
     */
    public void setCcRecipients(EmailAddress[] ccRecipients) {
        this.ccRecipients = ccRecipients;
    }

    /**
     * Returns the array of recipients
     *
     * @return bccRecipients array of bcc recipients
     * @since ARP 2.0
     */
    public EmailAddress[] getBccRecipients() {
        return bccRecipients;
    }

    /**
     * Set the array of recipients
     *
     * @param bccRecipients array of bcc recipients
     * @since ARP 2.0
     */
    public void setBccRecipients(EmailAddress[] bccRecipients) {
        this.bccRecipients = bccRecipients;
    }

    /**
     * Returns an array of attachments
     *
     * @return emailAttachmentData array with the email attachments
     * @since ARP 2.0
     */
    public EmailAttachmentData[] getEmailAttachmentData() {
        return emailAttachmentData;
    }

    /**
     * Set the email attachment data array
     *
     * @param emailAttachmentData array of email attatchments
     * @since ARP 2.0
     */
    public void setEmailAttachmentData(EmailAttachmentData[] emailAttachmentData) {
        this.emailAttachmentData = emailAttachmentData;
    }

    /**
     * Returns the message body of the email
     *
     * @return message Body string of the email
     * @since ARP 2.0
     */
    public String getMessageBody() {
        return messageBody;
    }

    /**
     * Set the message body of the email
     *
     * @param messageBody message body of the email
     * @since ARP 2.0
     */
    public void setMessageBody(String messageBody) {
        this.messageBody = messageBody;
    }

    /**
     * Returns the myme type of the message body
     *
     * @return mime type string of the message boddy
     * @since ARP 2.0
     */
    public String getMessageBodyMimeType() {
        return messageBodyMimeType;
    }

    /**
     * Set the mime type for the message body
     *
     * @param messageBodyMimeType type of the body message
     * @since ARP 2.0
     */
    public void setMessageBodyMimeType(String messageBodyMimeType) {
        this.messageBodyMimeType = messageBodyMimeType;
    }

    /**
     * Returns the subject of the email
     *
     * @return subject string of the email
     * @since ARP 2.0
     */
    public String getSubject() {
        return subject;
    }

    /**
     * Set the subject of the email
     *
     * @param subject of the email
     * @since ARP 2.0
     */
    public void setSubject(String subject) {
        this.subject = subject;
    }
}
